x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/rest/inference/RestStartTrainedModelDeploymentActionTests.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.rest.inference;
+
+import org.apache.lucene.util.SetOnce;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.test.rest.FakeRestRequest;
+import org.elasticsearch.test.rest.RestActionTestCase;
+import org.elasticsearch.xpack.core.ml.action.CreateTrainedModelAssignmentAction;
+import org.elasticsearch.xpack.core.ml.action.StartTrainedModelDeploymentAction;
+import org.elasticsearch.xpack.core.ml.inference.assignment.TrainedModelAssignment;
+import org.junit.Before;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.mockito.Mockito.mock;
+
+public class RestStartTrainedModelDeploymentActionTests extends RestActionTestCase {
+
+    boolean disableInferenceProcessCache = randomBoolean();
+
+    @Before
+    public void setUpAction() {
+        controller().registerHandler(new RestStartTrainedModelDeploymentAction(disableInferenceProcessCache));
+    }
+
+    public void testUsesDefaultTimeout() {
+        SetOnce<Boolean> executeCalled = new SetOnce<>();
+        verifyingClient.setExecuteVerifier(((actionType, actionRequest) -> {
+            assertThat(actionRequest, instanceOf(StartTrainedModelDeploymentAction.Request.class));
+
+            var request = (StartTrainedModelDeploymentAction.Request) actionRequest;
+            if (disableInferenceProcessCache) {
+                assertThat(request.getCacheSize(), is(ByteSizeValue.ZERO));
+            } else {
+                assertNull(request.getCacheSize());
+            }
+
+            executeCalled.set(true);
+            return createResponse();
+        }));
+
+        RestRequest inferenceRequest = new FakeRestRequest.Builder(xContentRegistry()).withMethod(RestRequest.Method.POST)
+            .withPath("_ml/trained_models/test_id/deployment/_start")
+            .build();
+        dispatchRequest(inferenceRequest);
+        assertThat(executeCalled.get(), equalTo(true));
+    }
+
+    private static CreateTrainedModelAssignmentAction.Response createResponse() {
+        return new CreateTrainedModelAssignmentAction.Response(mock(TrainedModelAssignment.class));
```

Comment:
Instead of mocking it and needing remove `final` can you use the tests helper? https://github.com/elastic/elasticsearch/blob/ad63465c6430b8ae8fb6dbefa441d1cd5d24b97d/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/CreateTrainedModelAssignmentActionResponseTests.java#L20

Or is it not accessible?

Reply:
Actually, it's protected...so instead you could create a public method in that class that exposes it or you can just do what it's doing:

```
return new Response(TrainedModelAssignmentTests.randomInstance());
```

Reply:
I hadn't even considered that the object's tests would have their own instantiation function. Thanks!